src/main/java/uk/gov/companieshouse/authcode/changed/service/AssociationService.java

```
@@ -0,0 +1,95 @@
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.authcode.changed.exceptions.InternalServerErrorRuntimeException;
+import uk.gov.companieshouse.authcode.changed.exceptions.NotFoundRuntimeException;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.authcode.changed.rest.AccountsAssociationEndpoint;
+import uk.gov.companieshouse.authcode.changed.utils.StaticPropertyUtil;
+
+@Service
+public class AssociationService {
+
+    public final AccountsAssociationEndpoint accountsAssociationEndpoint;
+
+    private static final Logger LOG = LoggerFactory.getLogger(StaticPropertyUtil.APPLICATION_NAMESPACE);
+
+    public AssociationService(AccountsAssociationEndpoint accountsAssociationEndpoint) {
+        this.accountsAssociationEndpoint = accountsAssociationEndpoint;
+    }
+
+    public void setCompanyAssociationToUnauthorised(String companyNumber)
+            throws ApiErrorResponseException, URIValidationException {
+        LOG.info("Setting association status to UNAUTHORISED for company number: " + companyNumber);
+        PrivateAccountsAssociationForCompanyGet associationsList = fetchAssociationDetails(companyNumber);
+        if (associationsList != null) {
+            associationsList.execute()
+                    .getData()
+                    .getItems()
+                    .forEach(association -> {
+                        String associationId = association.getId();
+                        LOG.info("Updating status for association ID: " + associationId);
+                        createUpdateStatusRequest(associationId, StatusEnum.UNAUTHORISED);
+                    });
+        } else {
+            LOG.error("No associations found for company number: " + companyNumber);
```

should we throw an error if there's no company found? instead of just logging?